src/dappsExplorer/utils.tsx

```diff
@@ -0,0 +1,31 @@
+import { DappV1WithCategoryName, DappV2WithCategoryNames, isDappV2 } from 'src/dapps/types'
+
+/***
+ * Calculates a score based on the search term and dapp
+ * - +2 if the name matches the search term in a case sensitive way
+ * - +1 if the name matches the search term in a case insensitive way
+ * - +1 if the description matches the search term in a case insensitive way
+ * - +1 if the category matches the search term in a case insensitive way
+ * @param dapp expects a dapp with a categoryName or categoryNames
+ * @param searchTerm
+ * @returns {number} the score of the dapp 0 - 5
+ */
+export const calculateSearchScore = (
+  dapp: DappV1WithCategoryName | DappV2WithCategoryNames,
```

Comment:
Do we need the search for v1 dapps?

Reply:
We do not this also simplifies some of the typing needed in #3599.